.github/workflows/build.yaml

```diff
@@ -20,12 +20,17 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 20
     permissions:
-      contents: read
-      packages: write
+       contents: read
+       packages: write
+       pull-requests: write
+       actions: write
+       statuses: write
+       checks: write
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Will this work on the `main` branch?

Reply:
should work based on the similar pattern we had for SonarQube checkouts: https://github.com/SolaceDev/maas-ep-core/blob/1f97c227869366fce9476f7e9bf1488c69d70e4f/.github/workflows/sonar-scan.yml#L15